src/main/java/com/es/phoneshop/web/CartPageServlet.java

```diff
@@ -0,0 +1,80 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.cart.CartService;
+import com.es.phoneshop.model.cart.HttpSessionCartService;
+import com.es.phoneshop.model.cart.OutOfStockException;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CartPageServlet extends HttpServlet {
+    private CartService cartService;
+
+    private Map<Long, String> errors;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        cartService = HttpSessionCartService.getInstance();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        request.setAttribute("cart", cartService.getCart(request).getCartItems());
+        request.setAttribute("errors", errors);
+        request.getRequestDispatcher("/WEB-INF/pages/cart.jsp").forward(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        List<Integer> quantities = new ArrayList<>();
+        List<Long> productIds = new ArrayList<>();
+        errors = new HashMap<>();
+
+        if (request.getParameterValues("productId").length == 0 && request.getParameterValues("quantity").length == 0) {
```

Comment:
What if one of the values will be empty and second - not empty? Should we also show any error message?